Sources/SmokeDynamoDB/InMemoryDynamoDBCompositePrimaryKeyTableStore.swift

```diff
@@ -167,49 +167,66 @@ internal class InMemoryDynamoDBCompositePrimaryKeyTableStore {
         return EventLoopFuture.andAllSucceed(futures, on: eventLoop)
     }
     
-    
     public func monomorphicBulkWriteWithoutThrowing<AttributesType, ItemType>(
         _ entries: [WriteEntry<AttributesType, ItemType>],
         eventLoop: EventLoop) async throws -> [Int: BatchStatementError] {
             var errors: [Int: BatchStatementError] = [:]
             
-            let futures = entries.enumerated().map { (index, entry) -> EventLoopFuture<Void> in
+            let futures = entries.enumerated().map { (index, entry) -> EventLoopFuture<Int?> in
                 switch entry {
                 case .update(new: let new, existing: let existing):
                     return updateItem(newItem: new, existingItem: existing, eventLoop: eventLoop)
-                        .flatMapError { err in
-                            errors[index] = BatchStatementError(code: .duplicateitem, message: nil)
-                            let promise = eventLoop.makePromise(of: Void.self)
-                            promise.succeed(())
+                        .flatMap { _ -> EventLoopFuture<Int?> in
+                            let promise = eventLoop.makePromise(of: Int?.self)
+                            promise.succeed(nil)
+                            return promise.futureResult
```

Comment:
For reference, this could be more simply written as

```
.map { _ -> Int? in
    return nil
```

Reply:
good point. updated!